modules/gapi/src/backends/fluid/gfluidcore.cpp

```diff
@@ -1295,32 +1313,85 @@ GAPI_FLUID_KERNEL(GFluidMulCOld, cv::gapi::core::GMulCOld, true)
     }
 };
 
-GAPI_FLUID_KERNEL(GFluidDivC, cv::gapi::core::GDivC, false)
+GAPI_FLUID_KERNEL(GFluidDivC, cv::gapi::core::GDivC, true)
 {
     static const int Window = 1;
 
-    static void run(const View &src, const cv::Scalar &_scalar, double _scale, int /*dtype*/,
-                    Buffer &dst)
+    static void run(const View& src, const cv::Scalar& _scalar, double _scale, int /*dtype*/,
+                    Buffer& dst, Buffer& scratch)
     {
-        const float scalar[4] = {
-            static_cast<float>(_scalar[0]),
-            static_cast<float>(_scalar[1]),
-            static_cast<float>(_scalar[2]),
-            static_cast<float>(_scalar[3])
-        };
-        const float scale = static_cast<float>(_scale);
+        GAPI_Assert(src.meta().chan <= 4);
+
+        if (dst.y() == 0)
+        {
+            const int chan = src.meta().chan;
+            float* sc = scratch.OutLine<float>();
+            int scratch_length = scratch.length();
+
+            scalar_to_scratch(_scalar, sc, scratch_length - 1, chan);
```

Comment:
Could rows above and below be part of `scalr_to_scratch` fucntion as well?
If you requre some specialization for different types is is possible to add specialization for float here in a way:

```
templae<class T>
scalar_to_scratch() {
for ()
auto = static_cast<T>(scalar)..
}
...
template<>
scalar_to_scratch<float>()
for ()
auto =...

if (std::fasb() < FLT_EPSILON) 
{
sc []= 1.0
}
```